docs/csharp/nullable-warnings.md

```diff
@@ -0,0 +1,130 @@
+
+The other actions instruct the compiler that the right-hand-side of the assignment is *not-null*. The expression on the right-hand-side could be null checked before assignment, as shown in the following example:
+
+:::code language="csharp" source="./snippets/null-warnings/Program.cs" id="NullGuard":::
+
+The previous examples demonstrate assignment to the return value of a method. You may annotate the method (or property) to indicate when a method returns a not-null value. The <xref:System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute?displayProperty=nameWithType> often specifies that a return value is *not-null* when an input argument is *not-null*. Another alternative is to add the null forgiving operator, `!` to the right-hand side:
+
+```csharp
+string msg = TryGetMessage(42)!;
+```
+
+Fixing a warning for assigning a *maybe-null* expression to a *not-null* variable involves one of four techniques:
+
+- Change the left side of the assignment to a nullable type. This action may introduce new warnings when you dereference that variable.
+- Provide a null-check before the assignment.
+- Annotate the API that produces the right-hand side of the assignment.
+- Add the null forgiving operator to the right-hand side of the assignment.
+
+## Nonnullable reference not initialized
+
+Other warnings are generated when a nonnullable reference variable isn't initialized when declared, or in a constructor. Consider the following class as an example:
+
+:::code language="csharp" source="./snippets/null-warnings/PersonExamples.cs" id="PersonExample":::
+
+Neither `FirstName` nor `LastName` are guaranteed initialized. If this code is new, consider changing the public interface. The above example could be updated as follows:
+
+:::code language="csharp" source="./snippets/null-warnings/PersonExamples.cs" id="WithConstructor":::
+
+If you require creating a `Person` object before setting the name, you can initialize the properties using a default non-null value:
+
+:::code language="csharp" source="./snippets/null-warnings/PersonExamples.cs" id="Initializer":::
+
+Another alternative may be to change those members to nullable reference types. The `Person` class could be defined as follows if `null` should be allowed for the name:
+
+:::code language="csharp" source="./snippets/null-warnings/PersonExamples.cs" id="NullableMember":::
+
+Existing code may require other changes to inform the compiler about the null semantics for those members. You may have created multiple constructors, and your class may have a private helper method that initializes one or more members. The <xref:System.Diagnostics.CodeAnalysis.MemberNotNullAttribute?displayProperty=nameWithType> and <xref:System.Diagnostics.CodeAnalysis.MemberNotNullWhenAttribute.%23ctor%2A?displayProperty=nameWithType> attributes inform the compiler that a member is *not-null* after the method has been called.
```

Comment:
I do not believe that you intended to have it render this way, but I see the fully qualified type and the type name.type name (which feels a bit redundant). I think the .ctor wasn't rendering correctly as an xref.

![image](https://user-images.githubusercontent.com/7679720/133834255-079f9b0e-8e72-4abb-b6b1-34541f0c9c82.png)